src/test/java/teammates/test/pageobjects/InstructorFeedbackEditPage.java

```diff
@@ -1554,24 +1554,58 @@ public void clickRemoveRankOptionLink(int qnIndex, int optionIndex) {
         click(removeOptionLink);
     }
 
+    public String getQuestionType(int qnNumber) {
+        return browser.driver.findElement(By.cssSelector("#form_editquestion-" + qnNumber + " input[name='questiontype']"))
+                .getAttribute("value").toString();
+    }
+
     public int getNumOfOptionsInRankOptions(int qnIndex) {
-        WebElement rankOptionsTable = browser.driver.findElement(By.id("rankOptionTable-" + qnIndex));
-        List<WebElement> optionInputFields = rankOptionsTable
-                                                .findElements(
-                                                     By.cssSelector("input[id^='rankOption-']"));
-        return optionInputFields.size();
+        if ("RANK_OPTIONS".equals(getQuestionType(qnIndex))) {
+            WebElement rankOptionsTable = browser.driver.findElement(By.id("rankOptionTable-" + qnIndex));
+            List<WebElement> optionInputFields = rankOptionsTable
+                                                    .findElements(
+                                                         By.cssSelector("input[id^='rankOption-']"));
+            return optionInputFields.size();
+        }
+
+        // For rank recipients
+        String recipient = getRecipientTypeForQuestion(qnIndex);
+
+        switch (recipient) {
+        case "STUDENTS":
+            return Integer.parseInt(browser.driver.findElement(By.id("num-students")).getAttribute("value").toString());
+        case "INSTRUCTORS":
+            return Integer.parseInt(browser.driver.findElement(By.id("num-instructors")).getAttribute("value").toString());
+        case "TEAMS":
+            return Integer.parseInt(browser.driver.findElement(By.id("num-teams")).getAttribute("value").toString());
+        case "OWN_TEAM_MEMBERS":
+            return Integer.MAX_VALUE;
+        case "OWN_TEAM_MEMBERS_INCLUDING_SELF":
+            return Integer.MAX_VALUE;
+        default:
+            // For SELF, OWN_TEAM, NONE
+            return 1;
```

Comment:
Here you see how my previous suggestion works. You can make it even more concise by allowing fall-throughs.